sycl/include/CL/sycl/half_type.hpp

```diff
@@ -400,6 +400,37 @@ class half {
     half r = *this;
     return -r;
   }
+
+// Operator +, -, *, /
+#define OP(op, op1)                                                            \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
+                                                const half &rhs) {             \
+    half rtn = lhs;                                                            \
+    rtn op1 rhs;                                                               \
+    return rtn;                                                                \
+  }                                                                            \
+  template <typename T,                                                        \
+            typename = std::enable_if_t<std::is_arithmetic<T>::value>>         \
+  __SYCL_CONSTEXPR_HALF friend half operator op(const half &lhs,               \
```

Comment:
How does this work if 'T' is double? Shouldn't ('half' + double') be 'double'?

Reply:
I agree it should return double for half+double.

Also I think this should work correctly for implicit conversion. If the other type is implicit convertible to an arithmetic type it should work too. Because that's how operators for arithmetic types work. I'm not aware of any way of achieving that with a template operator and I think you need the overloads:
```
half operator+(half, half);
float operator+(half, float);
double operator+(half, double);
half operator+(half, int);
half operator+(half, long);
half operator+(half, long long);
```
I did some testing to convince myself you can't do this with template and which overloads are needed (non-obvious for the integral types). You can find those tests here: https://godbolt.org/z/5GTzcfzP6.

Reply:
Yes, is should return double with half + double. I have changed to operators to return the appropriate types.
Thank you for the godbolt tests.